### PR TITLE
S3 Repository: Remove region setting

### LINKED ISSUE #22758: Remove either endpoint or region setting in aws cloud plugins
We have two different settings, `endpoint` and `region`, which determine the endpoint that is set on the aws client (and both of these appear unnecessary by default, as neither are required, and the s3 client appears to automatically find the region for a given bucket).  My understanding is that it is useful to have an explicit setting to allow connecting to new regions without needing a new release. However, I don't think we need both settings? We should settle on one (I would think endpoint, and remove region altogether?).

### PR DESCRIPTION
**Pull request by rjernst**
This change removes the ability to set region for s3 repositories.
Endpoint should be used instead if a custom s3 location needs to be
used.

closes #22758

### CODE REVIEW

**Review by dadoonet — APPROVED** [review]
It looks good. I just wonder why we added support for s3.client.*.region in the first place.

I believe we should not introduce an already deprecated setting in the first place, no?

**Comment by rjernst** [comment]
> I just wonder why we added support for s3.client.*.region in the first place.

I was not sure what the feedback would be at the time on removing region, and chose to follow up separately. We could remove it from 5.x, since it has not been released yet.

**Comment by dadoonet** [comment]
I see. Let's remove it then and change the doc then in this PR where we mention it.

File: docs/reference/migration/migrate_6_0/plugins.asciidoc [hunk 1]
```diff
@@ -14,3 +14,9 @@ It must exist before the s3 repository is created.
 * Support for specifying s3 credentials through environment variables and
 system properties has been removed. Use the `elasticsearch-keystore` tool
 to securely store the credentials.
+
+* Specifying region has been removed. This includes the settings `cloud.aws.region`,
+`cloud.aws.s3.region`, `repositories.s3.region`, `s3.client.*.region` and specifying
```
Comment (dadoonet) [inline docs/reference/migration/migrate_6_0/plugins.asciidoc]: So just remove here 

> , `s3.client.*.region`

**Pull request merged**